_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+
+To support each of the scenario above, `terragrunt` will need to be configured to know which subtemplates should be used
+for each of the scenarios, in addition to knowing where the `infrastructure-live` and `infrastructure-modules`
+repositories live. This should be configured in a new configuration file for Terragrunt, which will be autogenerated
+from an interactive setup process the first time you run `terragrunt new`.
+
+The following configuration options will be required:
+
+- `infrastructure_live_repo`: Git URL of the `infrastructure-live` repository.
+- `infrastructure_modules_repo`: Git URL of the `infrastructure-modules` repository.
+- `infrastructure_templates_repo`: Git URL of the `infrastructure-templates` repository. If you follow the convention,
+  this is the only necessary config to discover the templates for each of the scenarios listed above.
+- `new_environment_template_source`: Git URL OR local path to the template source for the `new environment` scenario.
+  If it is a relative path, will assume it is relative to the `infrastructure_templates_repo`. This can be used to
+  override the location of the template, as an alternative to the convention followed in `infrastructure-templates`, or
+  even source from a completely different repository.
+- `new_region_template_source`, `new_account_template_source`, `new_wrapper_module_template_source`: Similar to
+  `new_environment_template_source`, but for each of the other scenarios that `terragrunt new` supports.
+
+### Template repository structure
+
+The `new` command will assume the following template repository structure:
+
+```
+infrastructure-templates-example
+├── CONTRIBUTING.md
+├── LICENSE.txt
+├── README.md
+├── templates
+│   ├── account/
+│   ├── component/
+│   ├── environment/
+│   ├── region/
+│   └── wrapper-module/
+└── test
+```
+
+Where `templates` and `test` are directories that contain the template folders and corresponding tests, respectively.
+
+By convention, templates in the `infrastructure-templates` repo with the following names will automatically be used for
+each of the `new` command use cases:
+
+- `component` => `terragrunt new component`
+- `environment` => `terragrunt new environment`
+- `region` => `terragrunt new region`
+- `account` => `terragrunt new account`
+- `wrapper-module` => `terragrunt new wrapper-module`
+
+Note that the templates folder is not restricted to just these 4 templates: you are free to define as many or as little
+templates as you need. Additional templates that do not fit the convention can be invoked using `terragrunt new
+unicorn`.
+
+### Template engine
+
+Templates are written and rendered using the default Go Template engine. While there are many other template engines to
+consider, given that the templates are rendered using Terragrunt as a frontend, it makes the most sense to use the built
+in templating engine of Go.
```

Comment:
Hm, this requires us to:

1. Define our own format for asking users for inputs.
1. Use Go templating syntax in folder/file paths too, as those will need to be templated.
1. Add a lot of helper functions to work around the many limitations of Go templating...

Are there really no better templating languages that have native Go support these days?

Reply:
See https://github.com/gruntwork-io/terragrunt/blob/da1e3da8eb16ff29e58449c40f3ad98809c2436f/_docs/rfc/scaffolding_new_modules.md#alternative-template-engines